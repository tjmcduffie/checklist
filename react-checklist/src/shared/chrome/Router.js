import React from 'react';
import {BrowserRouter, Route, Switch} from 'react-router-dom';
import Deferred from '../base/Deferred';
import Nav from './Nav';
import routesConfig from '../../routesConfig';

const Router = () => (
  <BrowserRouter>
    <Nav />
    <Switch>
      <Route path="/" exact render={props => (<div>index</div>)} />
      {routesConfig.map(route => (
        <Route
          exact
          key={route.id}
          path={route.browserPath}
          render={
            props => (<Deferred importFn={route.importFn} id={route.id} />)
          }
        />
      ))}
    </Switch>
  </BrowserRouter>
);

export default Router;
