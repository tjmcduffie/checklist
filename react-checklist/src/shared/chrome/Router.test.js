import React from 'react';
import {create} from 'react-test-renderer';
import Router from './Router';

describe('Router', () => {
  const noop = () => {};

  it('renders without crashing', () => {
    const elem = create(<Router />);
    elem.unmount();
  });

  it('renders predictably', () => {
    const elem = create(<Router />);
    expect(elem.toJSON()).toMatchSnapshot();
  });
});
