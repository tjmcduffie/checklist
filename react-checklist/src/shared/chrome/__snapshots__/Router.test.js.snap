// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router renders predictably 1`] = `
Array [
  <nav>
    <ul>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
        <a
          href="/react-classic"
          onClick={[Function]}
        >
          React Classic
        </a>
        <a
          href="/react-hooks"
          onClick={[Function]}
        >
          React Hooks
        </a>
      </li>
    </ul>
  </nav>,
  <div>
    index
  </div>,
]
`;
